components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -122,6 +133,45 @@ static uint64_t _tick_get_ms(void)
     return esp_timer_get_time()/1000;
 }
 
+static esp_err_t esp_websocket_new_buf(esp_websocket_client_handle_t client, bool is_tx)
```

Comment:
We can avoid the boolean if we have in the call site:
```cpp
esp_websocket_new_buf(client->tx_buffer);
esp_websocket_new_buf(client->rx_buffer);
```

Reply:
We need `client` handle to know `client->buffer_size` value to allocate property size memory. Only tx_buffer or rx_buffer point is not enough.